sunpy/map/mapbase.py

```diff
@@ -2460,9 +2487,30 @@ def plot(self, annotate=True, axes=None, title=True,
                 norm.vmax = imshow_args.pop('vmax')
 
         if self.mask is None:
-            ret = axes.imshow(self.data, **imshow_args)
+            data = self.data
+        else:
+            data = np.ma.array(np.asarray(self.data), mask=self.mask)
+
+        if autoalign == 'pcolormesh':
+            axes.set_aspect(getattr(imshow_args, 'aspect', 1))
```

Comment:
I assume you mean this?
```suggestion
            axes.set_aspect(imshow_args.get('aspect', 1))
```

Reply:
Ha!  Yeah...